### PR TITLE
Docs clarification: local static variable

### PR DESCRIPTION
**Pull request by rofrol**
Second attempt after this https://github.com/ziglang/zig/pull/8375

Basing change on this:

>All global variables are static, but all static variables are not necessarily global.
>
>What all static variables (including global) have in common is that they have a lifetime that lasts the duration of the program. They are initialized just once, when the program begins.
>
>Local variables—which are “automatic” by default—are created each time a function is called, no matter how many times that function is on the stack… therefore they are appropriate for functions called recursively or functions involved with multiple threads. A local automatic variable is initialized each and every time the function is started.
>
>Local variables may also be declared static, in which case they work like global variables but have visibility limited to the function in which declared. Remember that a local static is like a global in that it is only initialized once
https://www.quora.com/What-is-the-difference-between-a-static-variable-and-global-variable-in-C-C++/answer/Brian-Overland-1

>In computer programming, a global variable is a variable with global scope, meaning that it is visible (hence accessible) throughout the program, unless shadowed.
https://en.wikipedia.org/wiki/Global_variable

>In computer programming, a static variable is a variable that has been allocated "statically", meaning that its lifetime (or "extent") is the entire run of the program. https://en.wikipedia.org/wiki/Static_variable

### CODE REVIEW

File: doc/langref.html.in [hunk 1]
```diff
@@ -906,14 +906,35 @@ test "namespaced global variable" {
...
 }
       {#code_end#}
+      <p>
+        Local static variables may also be declared inside a {#link|struct#}, {#link|union#}, or {#link|enum#}:
+        </p>
+        {#code_begin|test|namespaced_local_static#}
+  const std = @import("std");
+  const expect = std.testing.expect;
+  
+  test "namespaced localc static variable" {
```
Comment (g-w1) [inline doc/langref.html.in:925]: Typo?

**Comment by ifreund** [comment]
One point of inconsistency: what you are calling global variables aren't truly global either as they are scoped to the file.  Perhaps we should just get rid of the whole global/local thing and just call them static variables everywhere.

**Comment by rofrol** [comment]
Added commit that differentiate between static global and static local variable. Also uses file scope instead of global scope.

>Static global variable
A variable declared as static at the top level of a source file (outside any function definitions) is only visible throughout that file ("file scope", also known as "internal linkage"). In this usage, the keyword static is known as an "access specifier".
https://en.wikipedia.org/wiki/Static_(keyword)#Static_global_variable

**Comment by Mouvedia** [comment]
While you are at it can you replace "top-level" [here](https://github.com/ziglang/zig/blob/0963b76837d9da955637bd0988411340abf86ede/doc/langref.html.in#L5684)?

**Comment by rofrol** [comment]
@Mouvedia replace with what?

**Comment by Mouvedia** [comment]
> @Mouvedia replace with what?

I dunno the terminology in zig, sorry. I know it's wrong though, since it can be used in a struct.

**Comment by rofrol** [comment]
@Mouvedia Can you show some example code? I can't make it work inside struct.

**Comment by g-w1** [comment]
```zig
pub fn main() void {
    const L = struct {
        usingnamespace struct {
            pub fn f() void {}
        };
    };
    L.f();
}
```

File: doc/langref.html.in [hunk 2]
```diff
@@ -5681,14 +5681,24 @@ test "@intToPtr for pointer to zero bit type" {
...
+      {#syntax#}usingnamespace{#endsyntax#} is a declaration that imports all the public declarations of
       the operand, which must be a {#link|struct#}, {#link|union#}, or {#link|enum#}, into the current scope:
       </p>
       {#code_begin|test|usingnamespace#}
 usingnamespace @import("std");
 
 test "using std namespace" {
     testing.expect(true);
+
+    // Also inside struct
```
Comment (Mouvedia) [inline doc/langref.html.in:5693]: In the current documentation, each test is separated.
Here you are conflating two use cases in one test.

File: doc/langref.html.in [hunk 3]
```diff
@@ -5681,14 +5681,24 @@ test "@intToPtr for pointer to zero bit type" {
...
+    pub fn main() void {
```
Comment (Mouvedia) [inline doc/langref.html.in:5694]:
```
error: expected token '}', found 'pub'
    pub fn main() void {
    ^
```

**Review by Vexu — COMMENTED** [review]
I'm not a huge fan of calling them "Global static variables" as to me it implies the existence of "non-static global variables", maybe it would be enough to just call them "Static variables"?

**Pull request merged**